Src/Plugins/Security/SEC101_005.SlackTokenValidator.cs

```diff
@@ -66,40 +68,57 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(Dictionary<
                                                                 Dictionary<string, string> options,
                                                                 ref ResultLevelKind resultLevelKind)
         {
-            var client = new WebClient();
-            var data = new NameValueCollection();
+            const string uri = "https://slack.com/api/auth.test";
 
-            data["token"] = fingerprint.Secret;
+            try
+            {
+                HttpClient client = CreateOrUseCachedHttpClient();
+                var dict = new Dictionary<string, string>
+                {
+                    {"token", fingerprint.Secret },
+                };
 
-            byte[] bytes = client.UploadValues("https://slack.com/api/auth.test",
-                                                  "POST",
-                                                  data);
+                using var request = new HttpRequestMessage(HttpMethod.Post, uri);
+                request.Content = new FormUrlEncodedContent(dict);
 
-            string json = Encoding.UTF8.GetString(bytes);
+                using HttpResponseMessage response = client
+                    .SendAsync(request, HttpCompletionOption.ResponseHeadersRead)
+                    .GetAwaiter()
+                    .GetResult();
 
-            if (json.Contains("invalid_auth"))
-            {
-                return ValidationState.Unauthorized;
-            }
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    return ReturnUnexpectedResponseCode(ref message, response.StatusCode);
+                }
 
-            AuthTestResponse response = JsonSerializer.Deserialize<AuthTestResponse>(json);
+                string content = response.Content
+                                         .ReadAsStringAsync()
+                                         .GetAwaiter()
+                                         .GetResult();
 
-            switch (response.Error)
-            {
-                case "token_revoked":
-                case "account_inactive": { return ValidationState.Expired; }
-                case "invalid_auth": { return ValidationState.Unauthorized; }
-            }
+                AuthTestResponse authResponse = JsonSerializer.Deserialize<AuthTestResponse>(content);
 
-            if (!string.IsNullOrEmpty(response.Error))
+                switch (authResponse.Error)
+                {
+                    case "token_revoked":
+                    case "account_inactive": { return ValidationState.Expired; }
+                    case "invalid_auth": { return ReturnUnauthorizedAccess(ref message, fingerprint.Secret); }
```

Comment:
>ReturnUnauthorizedAccess [](http://example.com/codeflow?start=50&length=24)

dont do this. use validationState.unauthorized. If you do this, you are going to save the secret in the message.